### PR TITLE
HPCC-32366 Documentation for the Parquet Plugin Test Suite

### PR DESCRIPTION
**Pull request by ilhan2316**
## Type of change:
- [ ] This change is a bug fix (non-breaking change which fixes an issue).
- [ ] This change is a new feature (non-breaking change which adds functionality).
- [x] This change improves the code (refactor or other change that does not change the functionality)
- [ ] This change fixes warnings (the fix does not alter the functionality or the generated code)
- [ ] This change is a breaking change (fix or feature that will cause existing behavior to change).
- [ ] This change alters the query API (existing queries will have to be recompiled)

## Checklist:
- [x] My code follows the code style of this project.
  - [ ] My code does not create any new warnings from compiler, build system, or lint.
- [x] The commit message is properly formatted and free of typos.
  - [ ] The commit message title makes sense in a changelog, by itself.
  - [ ] The commit is signed.
- [ ] My change requires a change to the documentation.
  - [ ] I have updated the documentation accordingly, or...
  - [x] I have created a JIRA ticket to update the documentation.
  - [ ] Any new interfaces or exported functions are appropriately commented.
- [x] I have read the CONTRIBUTORS document.
- [ ] The change has been fully tested:
  - [ ] I have added tests to cover my changes.
  - [ ] All new and existing tests passed.
  - [ ] I have checked that this change does not introduce memory leaks.
  - [ ] I have used Valgrind or similar tools to check for potential issues.
- [ ] I have given due consideration to all of the following potential concerns:
  - [ ] Scalability
  - [ ] Performance
  - [ ] Security
  - [ ] Thread-safety
  - [ ] Cloud-compatibility
  - [ ] Premature optimization
  - [ ] Existing deployed queries will not be broken
  - [ ] This change fixes the problem, not just the symptom
  - [ ] The target branch of this pull request is appropriate for such a change.
- [x] There are no similar instances of the same problem that should be addressed
  - [ ] I have addressed them here
  - [x] I have raised JIRA issues to address them separately
- [ ] This is a user interface / front-end modification
  - [ ] I have tested my changes in multiple modern browsers
  - [ ] The component(s) render as expected

## Smoketest:
- [ ] Send notifications about my Pull Request position in Smoketest queue.
- [ ] Test my draft Pull Request.

### CODE REVIEW

File: testing/regress/ecl/README.md [hunk 1]
```diff
@@ -0,0 +1,128 @@
...
+- parquet-string.ecl: Focuses on string-related operations
+
+
+## Test Suite Overview
+# Type Testing
+
+Covers 42 data types including ECL and Arrow types
+Examples: BOOLEAN, INTEGER, STRING, UNICODE, various numeric types, sets, and Arrow-specific types
+
+ **Data Type Tests**
```
Comment (jackdelv) [inline testing/regress/ecl/README.md:61]: Should this be another header? Something like ### Data Type Tests.

File: testing/regress/ecl/README.md [hunk 2]
```diff
@@ -0,0 +1,128 @@
...
+- null
+- uint8
+- int8
+- uint16
+- int16
+- uint32
+- int32
+- uint64
+- int64
+- half
```
Comment (jackdelv) [inline testing/regress/ecl/README.md:93]: half_float?

File: testing/regress/ecl/README.md [hunk 3]
```diff
@@ -0,0 +1,128 @@
...
+- double
+- string
+- binary
+- fixed_size_binary
+- date32
+- date64
+- timestamp
+- time32
+- time64
+- interval_months
```
Comment (jackdelv) [inline testing/regress/ecl/README.md:104]: Are interval_months and interval_day_time supported? Are these in your tests?

File: testing/regress/ecl/README.md [hunk 4]
```diff
@@ -0,0 +1,128 @@
...
+- large_list
+- interval_day_time
+
+
+# Compression Testing
+
+Tests all available Arrow compression types: Snappy, GZip, Brotli, LZ4, ZSTD, Uncompressed
+Compares performance and file sizes for different compression options
+
+Parquet Read and Write Operations
```
Comment (jackdelv) [inline testing/regress/ecl/README.md:116]: Should this be a header?

File: testing/regress/ecl/README.md [hunk 5]
```diff
@@ -0,0 +1,128 @@
...
+
+Tests ParquetIO.Read for creating ECL datasets from Parquet files
+Tests ParquetIO.Write for writing ECL datasets to Parquet files
+
+# Additional Tests
+
+Spray and despray speeds comparison with other file types
```
Comment (jackdelv) [inline testing/regress/ecl/README.md:123]: In this context its better to say read and write speeds. We are also trying to move away from the terminology spray/despray.

File: testing/regress/ecl/README.md [hunk 6]
```diff
@@ -0,0 +1,128 @@
...
+Schema handling and compatibility
+Behavior with corrupt data and empty Parquet files
+
+# Test Evaluation
```
Comment (jackdelv) [inline testing/regress/ecl/README.md:127]: This subsection may not be necessary, or it should explain that some Parquet tests don't use key files. Also, the extension of the key files is ".xml".

**Review by jackdelv — CHANGES_REQUESTED** [review]
@ilhan2316 A few comments and questions.

File: testing/regress/ecl/README.md (hunk 3 above)
Comment (ilhan2316) [reply testing/regress/ecl/README.md:104]: Those are both in the data type test file in the suite.

File: testing/regress/ecl/README.md [hunk 7]
```diff
@@ -0,0 +1,200 @@
...
+
+- BOOLEAN: True/False values.
+- INTEGER: Signed integer values.
+- UNSIGNED: Unsigned integer values.
+- REAL: Floating-point values.
+- DECIMAL: Decimal values with specified precision.
+- STRING: Fixed-length strings.
+- DATA: Binary data converted to STRING.
+- VARSTRING: Variable-length strings.
+- QSTRING: Quick strings.
```
Comment (jackdelv) [inline testing/regress/ecl/README.md:144]: I'm not sure the Q in QString stands for Quick. QStrings are a simpler encoding of a limited set of characters. I'm not quite sure what the purpose of these descriptions are. Is it to describe the type or how that type is tested? In either case I think it requires more information in some cases and a second look.

**Review by jackdelv — CHANGES_REQUESTED** [review]
@ilhan2316 I think it is very close. One last question.

**Review by jackdelv — APPROVED** [review]
@ilhan2316 Looks good to me.

**Review by AttilaVamos — APPROVED** [review]
Seems good.

**Comment by ghalliday** [comment]
The commit message does not have the correct format - an extra space - which prevents it being picked up by jira bot.  I will merge anyway.

**Pull request merged**